pkg/scheduler/queue/tree_queue_algo_querier_worker_queue_priority.go

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package queue
+
+type QuerierWorkerQueuePriorityAlgo struct {
+	currentQuerierWorker  int
+	currentNodeOrderIndex int
+	nodeOrder             []string
+	nodeCounts            map[string]int
```

Comment:
What's the purpose of this? Like what do we read this for?

Reply:
this is for the equivalent of the `tenantNodes` in TQA - namely for when the algorithm is not at the root of the tree and therefore than can be multiple of each node type, we track that we don't need to add a node to the rotation if an instance of that node type already exists somewhere, and we shouldn't delete it from the rotation until the final instance is removed - I plan to add more docstring on this, but there is an full unit test for the behavior in this PR.

Reply:
Do we ever expect this algo to not be at the root of the tree? i.e., is this more than we realistically need to keep track of?

Reply:
I was under the impression that we would eventually make it so `use-multi-algorithm-tree` would use this algo regardless, and `prioritize-query-components` would put it at the first level.
Whether or not there is a config option placing this at a non-root level of the tree moving forward, I am not comfortable really having one of these algorithms in the codebase that breaks if it's not put at the correct level.

The breakage scenario is really bad too - prematurely deleting a node type from the rotation means it would not be selected and all queries sitting in that queue would be ignored & unreachable indefinitely until another enqueue of that node type re-adds it to the broken rotation tracking.

Reply:
Long term I do think we should limit the number of ways to configure it to what we use in production. Once that is the case, I think it would be simpler and likely faster to remove `nodeCounts` and just use `slices.Contains(nodeOrder, name)` instead. The map is expected to only have 4 elements - searching a slice of that size is faster than a map lookup.